packages/smooth_app/lib/generic_lib/buttons/smooth_large_button_with_icon.dart

```diff
@@ -1,24 +1,28 @@
 import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/data_models/user_preferences.dart';
 import 'package:smooth_app/generic_lib/buttons/smooth_simple_button.dart';
+import 'package:smooth_app/themes/theme_provider.dart';
 
 class SmoothLargeButtonWithIcon extends StatelessWidget {
   const SmoothLargeButtonWithIcon({
     required this.text,
     required this.icon,
     required this.onPressed,
-    required this.isDarkMode,
     this.padding,
   });
 
   final String text;
   final IconData icon;
   final VoidCallback onPressed;
-  final bool isDarkMode;
   final EdgeInsets? padding;
 
   @override
   Widget build(BuildContext context) {
     final ThemeData themeData = Theme.of(context);
+    final UserPreferences userPreferences =
+        Provider.of<UserPreferences>(context, listen: false);
+    final bool isDarkMode = ThemeProvider(userPreferences).isDarkMode(context);
```

Comment:
Just one thing I just noticed. The ThemeProvider is a provider, so you shouldn't create it all over again but directly access it through it